components/authz-service/storage/v2/postgres/postgres_test.go

```diff
@@ -3508,6 +3508,163 @@ func TestListRules(t *testing.T) {
+				[]string{"chef-server-2"}, storage.ChefServer, storage.MemberOf)
+			rule2, err := storage.NewRule("new-id-2", projID2, "name2", ruleType,
+				[]storage.Condition{condition4})
+			require.NoError(t, err)
+			_, err = store.CreateRule(ctx, &rule2)
+			require.NoError(t, err)
+
+			condition5, err := storage.NewCondition(ruleType,
+				[]string{"chef-server-3", "chef-server-4"}, storage.ChefServer, storage.MemberOf)
+			rule3, err := storage.NewRule("new-id-3", projID2, "name3", ruleType,
+				[]storage.Condition{condition5})
+			require.NoError(t, err)
+			_, err = store.CreateRule(ctx, &rule3)
+			require.NoError(t, err)
+			assertCount(t, 1, db.QueryRow(`SELECT count(*) FROM iam_project_rules WHERE id=$1`, rule1.ID))
+			assertCount(t, 1, db.QueryRow(`SELECT count(*) FROM iam_project_rules WHERE id=$1`, rule2.ID))
+			assertCount(t, 1, db.QueryRow(`SELECT count(*) FROM iam_project_rules WHERE id=$1`, rule3.ID))
+			assertCount(t, 5, db.QueryRow(`SELECT count(*) FROM iam_rule_conditions`))
+
+			resp, err := store.ListRulesForProject(ctx, projID2)
+			assert.NoError(t, err)
+			assert.Equal(t, 2, len(resp))
+			assert.ElementsMatch(t, []*storage.Rule{&rule2, &rule3}, resp)
+		},
+		"when the project is not in the filter, returns an empty list": func(t *testing.T) {
+			ctx := context.Background()
+
+			projID := "project-1"
+			insertTestProject(t, db, projID, "let's go jigglypuff - topsecret", storage.Custom)
+			projID2 := "project-2"
+			insertTestProject(t, db, projID2, "pika p", storage.Custom)
+			ctx = insertProjectsIntoContext(ctx, []string{"project-3", "project-4"})
+
+			ruleType := storage.Node
+			rule1 := createRuleWithMultipleConditions(t, store, projID, ruleType)
+
+			condition4, err := storage.NewCondition(ruleType,
+				[]string{"chef-server-2"}, storage.ChefServer, storage.MemberOf)
+			rule2, err := storage.NewRule("new-id-2", projID2, "name2", ruleType,
+				[]storage.Condition{condition4})
+			require.NoError(t, err)
+			_, err = store.CreateRule(ctx, &rule2)
+			require.NoError(t, err)
+
+			condition5, err := storage.NewCondition(ruleType,
+				[]string{"chef-server-3", "chef-server-4"}, storage.ChefServer, storage.MemberOf)
+			rule3, err := storage.NewRule("new-id-3", projID2, "name3", ruleType,
+				[]storage.Condition{condition5})
+			require.NoError(t, err)
+			_, err = store.CreateRule(ctx, &rule3)
+			require.NoError(t, err)
+			assertCount(t, 1, db.QueryRow(`SELECT count(*) FROM iam_project_rules WHERE id=$1`, rule1.ID))
+			assertCount(t, 1, db.QueryRow(`SELECT count(*) FROM iam_project_rules WHERE id=$1`, rule2.ID))
+			assertCount(t, 1, db.QueryRow(`SELECT count(*) FROM iam_project_rules WHERE id=$1`, rule3.ID))
+			assertCount(t, 5, db.QueryRow(`SELECT count(*) FROM iam_rule_conditions`))
+
+			resp, err := store.ListRulesForProject(ctx, projID2)
+			assert.NoError(t, err)
+			assert.Equal(t, 0, len(resp))
+			assert.ElementsMatch(t, []*storage.Rule{}, resp)
+		},
+	}
+
```

Comment:
👀 no shuffle? Hm. I think any test suite benefits from some controlled randomness 😉

Reply:
Missed that! Will add.